### PR TITLE
[Component with its own class name as service type] LPS-193631 Refactor ModelResourcePermission components

### PR DESCRIPTION
**Pull request by MrJohn1911**
Ticket: https://liferay.atlassian.net/browse/LPS-193631

### CODE REVIEW

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:data-engine

File: modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java [hunk 1]
```diff
@@ -1692,8 +1693,10 @@ private void _validate(
 	private static final EntityModel _entityModel =
 		new DataDefinitionEntityModel();
 
-	@Reference
-	private DataDefinitionModelResourcePermission
+	@Reference(
+		target = "(model.class.name=com.liferay.dynamic.data.mapping.model.DataDefinition)"
+	)
```
Comment (rafaprax) [inline modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/resource/v2_0/DataDefinitionResourceImpl.java:1698]: @MrJohn1911  use `com.liferay.data.engine.rest.dto.v2_0.DataDefinition` and change the previous changed reference to use the DTO class as well.

**Review by rafaprax — CHANGES_REQUESTED** [review]

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:data-engine

**Comment by catarineaguiar** [comment]
@MrJohn1911 regression found:

```
     [exec] com.liferay.data.engine.rest.resource.v2_0.test.DataRecordResourceTest > testGraphQLDeleteDataRecord FAILED
     [exec]     java.lang.AssertionError
     [exec]     	at org.junit.Assert.fail(Assert.java:87)
     [exec]     	at org.junit.Assert.assertTrue(Assert.java:42)
     [exec]     	at org.junit.Assert.assertTrue(Assert.java:53)
     [exec]     	at com.liferay.data.engine.rest.resource.v2_0.test.BaseDataRecordResourceTestCase.testGraphQLDeleteDataRecord(BaseDataRecordResourceTestCase.java:839)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
     [exec]     	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
     [exec]     	at java.lang.reflect.Method.invoke(Method.java:498)
     [exec]     	at com.liferay.arquillian.extension.junit.bridge.server.TestExecutorRunnable$3.evaluate(TestExecutorRunnable.java:344)
```
```
     [exec] com.liferay.data.engine.rest.resource.v2_0.test.DataRecordResourceTest > testGraphQLGetDataRecord FAILED
     [exec]     java.lang.IllegalArgumentException: Expected starts with '{', but found 'n' in 'null'
     [exec]     	at com.liferay.data.engine.rest.client.json.BaseJSONParser._assertStartsWithAndEndsWith(BaseJSONParser.java:249)
     [exec]     	at com.liferay.data.engine.rest.client.json.BaseJSONParser.parseToDTO(BaseJSONParser.java:40)
     [exec]     	at com.liferay.data.engine.rest.client.serdes.v2_0.DataRecordSerDes.toDTO(DataRecordSerDes.java:29)
     [exec]     	at com.liferay.data.engine.rest.resource.v2_0.test.BaseDataRecordResourceTestCase.testGraphQLGetDataRecord(BaseDataRecordResourceTestCase.java:894)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
     [exec]     	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
```
full log [here](https://testray.liferay.com/reports/production/logs/2023-08/test-1-40/test-portal-acceptance-pullrequest(master)/5263/modules-integration-mysql57-jdk8/0/2/jenkins-console.txt.gz?authuser=0)

**Comment by MrJohn1911** [comment]
ci:test:sf

**Comment by MrJohn1911** [comment]
ci:test:relevant

**Comment by MrJohn1911** [comment]
ci:test:startup

**Comment by MrJohn1911** [comment]
ci:test:data-engine

**Comment by Kvale1733** [comment]
ci:test:stable

**Comment by Kvale1733** [comment]
@MrJohn1911 No regressions found.

**Comment by shuyangzhou** [comment]
@MrJohn1911 Please inline those ModelResourcePermission into their ResourceImpl to avoid adding the weird model.class.name property.

CC @rafaprax

**Pull request closed without merging**